torchao/dtypes/float8/float8_layout.py

```diff
@@ -209,19 +215,64 @@ def __repr__(self):
         )
 
 
+class Float8QuantizedTensor(AffineQuantizedTensor):
```

Comment:
I'm not a fan of this, this introduces one more abstraction (`Float8QuantizedTensor`), while keeping the complexity of `AffineQuantizedTensor`.  I think either staying with AQT or just writing a float8 tensor without using AQT would seem more attractive.

Reply:
Interesting perspective - cc @jainapurva @jerryzh168 thoughts on this?